### PR TITLE
Cleanups

### LINKED ISSUE #142: Downgrading numpy from defaults -> conda-forge results in broken install
Issue:

If one installs numpy=1.16.x from defaults, and then downgrades to numpy=1.15.4 from conda-forge, numpy-base is not removed, resulting in a broken installation.

To reproduce:

```pytb
$ conda create --name nptest python=2.7 gwpy
$ conda install --name nptest numpy=1.15.4
$ conda activate nptest
$ python -c "import numpy"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/duncan/opt/miniconda3/envs/nptest/lib/python2.7/site-packages/numpy/__init__.py", line 142, in <module>
    from . import core
  File "/Users/duncan/opt/miniconda3/envs/nptest/lib/python2.7/site-packages/numpy/core/__init__.py", line 91, in <module>
    raise ImportError(msg.format(path))
ImportError: Something is wrong with the numpy installation. While importing we detected an older version of numpy in ['/Users/duncan/opt/miniconda3/envs/nptest/lib/python2.7/site-packages/numpy']. One method of fixing this is to repeatedly uninstall numpy until none is found, then reinstall this version.
```

I understand that this is a fairly niche error that can be resolved by manually forcing numpy from conda-forge in the first place, but I'm just wondering, what's the best way to avoid this in general?

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /Users/duncan/opt/miniconda3/envs/nptest:
#
# Name                    Version                   Build  Channel
asn1crypto                0.24.0                py27_1003    conda-forge
astropy                   2.0.8                    py27_0    conda-forge
atomicwrites              1.3.0                      py_0    conda-forge
attrs                     19.1.0                     py_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.5                        py_1    conda-forge
backports_abc             0.5                        py_1    conda-forge
blas                      1.1                    openblas    conda-forge
ca-certificates           2018.11.29           ha4d7672_0    conda-forge
certifi                   2018.11.29            py27_1000    conda-forge
cffi                      1.12.2           py27h342bebf_0    conda-forge
cryptography              2.5              py27hc2b1221_1    conda-forge
cycler                    0.10.0                     py_1    conda-forge
dqsegdb2                  1.0.1                      py_0    conda-forge
enum34                    1.1.6                 py27_1001    conda-forge
freetype                  2.9.1             h597ad8a_1005    conda-forge
funcsigs                  1.0.2                      py_3    conda-forge
functools32               3.2.3.2                    py_3    conda-forge
futures                   3.2.0                 py27_1000    conda-forge
gwdatafind                1.0.4                      py_0    conda-forge
gwosc                     0.4.1                      py_0    conda-forge
gwpy                      0.14.0                   py27_0    conda-forge
h5py                      2.9.0           nompi_py27h35dc34b_1102    conda-forge
hdf5                      1.10.4          nompi_h646315f_1105    conda-forge
idna                      2.8                   py27_1000    conda-forge
intel-openmp              2019.1                      144    defaults
ipaddress                 1.0.22                     py_1    conda-forge
kiwisolver                1.0.1           py27h04f5b5a_1002    conda-forge
libcxx                    7.0.0                h2d50403_1    conda-forge
libffi                    3.2.1             h0a44026_1005    conda-forge
libgfortran               3.0.1                         0    conda-forge
libopenblas               0.3.3                hdc02c5d_3    defaults
libpng                    1.6.36            ha441bb4_1000    conda-forge
ligo-common               1.0.3                      py_0    conda-forge
ligo-segments             1.2.0            py27h1de35cc_0    conda-forge
ligotimegps               1.2.3                      py_0    conda-forge
linecache2                1.0.0                      py_1    conda-forge
llvm-meta                 7.0.0                         0    conda-forge
matplotlib                2.2.3                    py27_1    conda-forge
matplotlib-base           2.2.3            py27hd64abcd_1    conda-forge
mkl                       2019.1                      144    defaults
mkl_fft                   1.0.10                   py27_0    conda-forge
mkl_random                1.0.2                    py27_0    conda-forge
more-itertools            4.3.0                 py27_1000    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
numpy                     1.15.4          py27_blas_openblash4251c03_1000  [blas_openblas]  conda-forge
numpy-base                1.16.2           py27ha711998_0    defaults
openblas                  0.3.3             hdc02c5d_1001    conda-forge
openssl                   1.1.1b               h1de35cc_0    conda-forge
pathlib2                  2.3.3                 py27_1000    conda-forge
pip                       19.0.3                   py27_0    conda-forge
pluggy                    0.9.0                      py_0    conda-forge
py                        1.8.0                      py_0    conda-forge
pycparser                 2.19                       py_0    conda-forge
pyopenssl                 19.0.0                   py27_0    conda-forge
pyparsing                 2.3.1                      py_0    conda-forge
pytest                    4.3.0                    py27_0    conda-forge
python                    2.7.15            h932b40d_1008    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
pytz                      2018.9                     py_0    conda-forge
readline                  7.0               hcfe32e1_1001    conda-forge
scandir                   1.9.0           py27h1de35cc_1000    conda-forge
scipy                     1.2.1           py27_blas_openblash486cb9f_0  [blas_openblas]  conda-forge
setuptools                40.8.0                   py27_0    conda-forge
singledispatch            3.4.0.3               py27_1000    conda-forge
six                       1.12.0                py27_1000    conda-forge
sqlite                    3.26.0            h1765d9f_1000    conda-forge
subprocess32              3.5.3            py27h470a237_0    conda-forge
tk                        8.6.9             ha441bb4_1000    conda-forge
tornado                   5.1.1           py27h1de35cc_1000    conda-forge
tqdm                      4.31.1                     py_0    conda-forge
traceback2                1.4.0                    py27_0    conda-forge
unittest2                 1.1.0                      py_0    conda-forge
wheel                     0.33.1                   py27_0    conda-forge
zlib                      1.2.11            h1de35cc_1004    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

     active environment : nptest
    active env location : /Users/duncan/opt/miniconda3/envs/nptest
            shell level : 3
       user config file : /Users/duncan/.condarc
 populated config files : /Users/duncan/opt/miniconda3/.condarc
                          /Users/duncan/.condarc
          conda version : 4.6.7
    conda-build version : 3.17.8
         python version : 3.7.2.final.0
       base environment : /Users/duncan/opt/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/osx-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/duncan/opt/miniconda3/pkgs
                          /Users/duncan/.conda/pkgs
       envs directories : /Users/duncan/opt/miniconda3/envs
                          /Users/duncan/.conda/envs
               platform : osx-64
             user-agent : conda/4.6.7 requests/2.21.0 CPython/3.7.2 Darwin/18.0.0 OSX/10.14
                UID:GID : 503:20
             netrc file : None
           offline mode : False

```
</details>

### LINKED ISSUE #160: Add run_constrained for `numpy-base 9999`
To avoid numpy from conda-forge and numpy-base from defaults being installed at the same time.

### PR DESCRIPTION
**Pull request by h-vetinari**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

The pypy builds on aarch/ppc failed several times for #233 (so new PR to fix CI seemed in order), plus I found some possibilities for clean-ups while working on #227.

Finally, there's a long-standing issues to deal better with the interaction of `numpy-base` from the anaconda default channels.
Fixes #142 
Fixes #160 
Might also solve #230

### CODE REVIEW

**Comment by h-vetinari** [comment]
Pypy on travis ran through, but was cutting it very close with 30 seconds to spare; let's see if some mild parallelism in the test suite can help...

**Comment by h-vetinari** [comment]
@rgommers, I fixed the pypy timeouts from #233 by skipping some of the longest-running tests (only for pypy on drone & travis). This also brought the runtime back down to having some breathing room (~20min) regarding the timeouts.

Note, this PR also (finally) adds `run_constrained` (cf. linked issues in OP). I think these cleanups are good to go in one, but can split into more PRs if really desired (e.g. skips, run_constrained, rest).

**Review by rgommers — COMMENTED** [review]
Thanks @h-vetinari. Test skips look fine to me. I'm not familiar with `run_constrained`, so would be good to get input from @isuruf or another core member on that.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -29,6 +29,9 @@ requirements:
     - liblapack
   run:
     - python
+  run_constrained:
+    # enforce eviction of package from anaconda defaults
+    - numpy-base 9999
```
Comment (xhochy) [inline recipe/meta.yaml:34]: This is more restrictive as `0a0` should be near to the minimal version conda can digest.
```suggestion
    - numpy-base <0a0
```

**Comment by xhochy** [comment]
We should also add that `run_constrained` as a repodata patch to all existing `numpy` packages.

File: recipe/meta.yaml (hunk 1 above)
Comment (h-vetinari) [reply recipe/meta.yaml:34]: Numpy 9999 is pretty unrealistic already, no? 😅 
And it's in line with other feedstocks, e.g. [pypy](https://github.com/conda-forge/pypy3.6-feedstock/blob/master/recipe/meta.yaml#L67-L69) resp. what @isuruf suggested in the title of #160

That said, I don't care much either way whether it's 9999, 99999999999, or <0a0...

**Comment by h-vetinari** [comment]
> 
> 
> We should also add that `run_constrained` as a repodata patch to all existing `numpy` packages.

The pain seems to have been manageable thus far (issues have been open for >2 years) - I see your point, and think it's a valid approach, but don't we also run some risks of breaking currently shaky-but-running environments that mix in `numpy-base`?

**Comment by h-vetinari** [comment]
Ping @isuruf

**Pull request merged**